File: BackEnd/tourr/src/main/java/com/idb/tour/service/BookingService.java
package com.idb.tour.service;

import java.util.List;

import com.idb.tour.model.Booking;

public interface BookingService {
    Booking getBookingById(Long id);
    List<Booking> getAllBookings();
    Booking saveBooking(Booking booking);
    void deleteBooking(Long id);
    Booking updateBookingStatus(Long id, String status); // New method to update status
    Booking updateBooking(Long id, Booking booking);
}

 

File: FrontEnd/travela/src/app/booking/booking.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { apiUrl } from '../constants';

@Injectable({
  providedIn: 'root'
})
export class BookingService {

  constructor(private http: HttpClient) { }

  getAllTours(): Observable<any> {
    return this.http.get(`${apiUrl}/api/tours`);
  }

  saveBooking(booking: any): Observable<any> {
    return this.http.post(`${apiUrl}/api/booking`, booking);
  }

  getAllBookings(): Observable<any> {
    return this.http.get(`${apiUrl}/api/bookings`);
  }

  getBookingById(id: number): Observable<any> {
    return this.http.get(`${apiUrl}/api/booking/${id}`);
  }

  updateBooking(booking: any): Observable<any> {
    return this.http.put(`${apiUrl}/api/bookings/${booking.id}`, booking);
  }
  updateBookingStatus(id: number, status: string): Observable<any> {
    return this.http.patch(`${apiUrl}/api/bookings/${id}/status`, null,{ params: { status } });
  }
}
